services/slam/collector.go

```diff
@@ -0,0 +1,52 @@
+package slam
+
+import (
+	"context"
+
+	pb "go.viam.com/api/service/slam/v1"
+	"google.golang.org/protobuf/types/known/anypb"
+
+	"go.viam.com/rdk/data"
+	"go.viam.com/rdk/spatialmath"
+)
+
+type method int64
+
+const (
+	getPosition      method = iota
+	getPointCloudMap method = iota
+)
+
+func (m method) String() string {
+	switch m {
+	case getPosition:
+		return "GetPosition"
+	case getPointCloudMap:
+		return "GetPointCloudMap"
+	}
+	return "Unknown"
+}
+
+func newGetPositionCollector(resource interface{}, params data.CollectorParams) (data.Collector, error) {
+	slam, err := assertSLAM(resource)
+	if err != nil {
+		return nil, err
+	}
+
+	cFunc := data.CaptureFunc(func(ctx context.Context, _ map[string]*anypb.Any) (interface{}, error) {
+		pose, componentRef, err := slam.GetPosition(ctx)
+		if err != nil {
+			return nil, data.FailedToReadErr(params.ComponentName, getPosition.String(), err)
+		}
+		return &pb.GetPositionResponse{Pose: spatialmath.PoseToProtobuf(pose), ComponentReference: componentRef}, nil
```

Comment:
Checking that this gets converted into a structpb as expected? This can be tested by locally printing what's saved in `getAndPushNextReading` or, even more ideally, connecting this to `app` and checking that it syncs the data that's formatted as expected to the debug sensor tab.

Reply:
from `getAndPushNextReading`
```
metadata:{time_requested:{seconds:1689109487  nanos:671327000}  time_received:{seconds:1689109487  nanos:819477000}}  struct:{fields:{key:"pose"  value:{struct_value:{fields:{key:"o_z"  value:{number_value:0.9999999999999999}}  fields:{key:"theta"  value:{number_value:0.08582535559390861}}  fields:{key:"x"  value:{number_value:5.921536787524186}}  fields:{key:"y"  value:{number_value:13.296696037491637}}}}}}
```